diff --git a/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/api-doc/ajax/get.jag b/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/api-doc/ajax/get.jag
index e9af331579..4330bf53f2 100644
--- a/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/api-doc/ajax/get.jag
+++ b/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/blocks/api-doc/ajax/get.jag
@@ -104,21 +104,6 @@ var log = new Log();
         var store = jagg.module("manager").getAPIStoreObj();
         gatewayURL = store.getHTTPsGatewayEndpointURLs();
 
-
-
-        //populate the security definitions
-        var default_implicit = {
-            "type":"oauth2",
-            "authorizationUrl": gatewayURL[0]+"/authorize",
-            "flow": "implicit",
-            "scopes":{}
-        };
-        if(jsonObj["securityDefinitions"] == undefined){
-            jsonObj["securityDefinitions"] = {};
-        }
-        jsonObj["securityDefinitions"]["default"] = default_implicit;
-
-        
         if(jsonObj["x-wso2-security"]!= undefined){
             var scopeList = jsonObj["x-wso2-security"]["apim"]["x-wso2-scopes"];
             for(var i =0; i < scopeList.length;i++){
@@ -163,6 +148,17 @@ var log = new Log();
            
             }
         }
+            //populate the security definitions
+            var default_implicit = {
+                "type":"oauth2",
+                "authorizationUrl": urlArr[0]+"/authorize",
+                "flow": "implicit",
+                "scopes":{}
+            };
+            if(jsonObj["securityDefinitions"] == undefined){
+                jsonObj["securityDefinitions"] = {};
+            }
+            jsonObj["securityDefinitions"]["default"] = default_implicit;
 
         jsonObj.basePath = api.context;
 
diff --git a/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/swagger/template.jag b/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/swagger/template.jag
index e7ccada967..8bc3245c89 100644
--- a/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/swagger/template.jag
+++ b/features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/wso2/templates/api/swagger/template.jag
@@ -218,6 +218,7 @@ var select_environment = function(){
     }else{
     window.swaggerUi.updateSwaggerUi({ "url" : swaggerUi.api.url + "?envName="+name});
     }
+    $("#swagger-api").attr("href", window.swaggerUi.api.url);
     change_token();
   };
 
@@ -355,7 +356,7 @@ function checkOnKeyPress(e) {
 <div class="col-md-12 col-lg-12">
     <div class="swagger-section">
     <div id="message-bar" class="swagger-ui-wrap">&nbsp;</div>
-    <div id="Download" style="float:right;font-size:14px"><a title="swagger" href='<%=encode.forHtmlAttribute(swaggerAPI)%>' target="_blank" style="font-size:14px"><i class="icon-share-alt" ></i> Swagger ( /swagger.json )</a></div>
+    <div id="Download" style="float:right;font-size:14px"><a id="swagger-api" title="swagger" href='<%=encode.forHtmlAttribute(swaggerAPI)%>' target="_blank" style="font-size:14px"><i class="icon-share-alt" ></i> Swagger ( /swagger.json )</a></div>
     <div id="swagger-ui-container" class="swagger-ui-wrap"></div>
 </div>
 </div>
